diff --git a/include/libnsfb.h b/include/libnsfb.h
index 674df65..c0a4298 100644
--- a/include/libnsfb.h
+++ b/include/libnsfb.h
@@ -41,6 +41,7 @@ enum nsfb_type_e {
     NSFB_SURFACE_VNC, /**< VNC surface */
     NSFB_SURFACE_LINUX, /**< Linux framebuffer surface */
     NSFB_SURFACE_ABLE, /**< ABLE framebuffer surface */
+    NSFB_SURFACE_VITA, /**< PlayStation Vita display surface (VitaSurf) */
     NSFB_SURFACE_RAM, /**< RAM surface */
     NSFB_SURFACE_COUNT, /**< The number of surface kinds */
 };
